src/agents/workflows/subgraphs/step4_features/solution/pdf_utils.py

```
@@ -0,0 +1,318 @@
+    return dict(envs or {})
+
+def _execute_pdf_locally(code: str, envs: dict) -> Tuple[bool, List[str], List[str], Optional[Exception]]:
+    """로컬에서 PDF 생성 코드 실행"""
+    temp_file = None
+    try:
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False, encoding='utf-8') as f:
+            f.write(code)
+            temp_file = f.name
+        
+        # [구조 개선] 바이너리 모드로 실행하여 인코딩 오류 원천 차단
+        result = subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)